src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs

```diff
@@ -454,27 +497,34 @@ private async Task ProcessIncomingAsync(ConnectionContext connection)
                 }
             }
             finally
-            {
-                keepAliveTimer.Stop();
-            }
+            {
+                keepAliveTimer.Stop();
+                Status = ServiceConnectionStatus.Disconnected;
+                await DisposeConnection(connection);
```

Comment:
This should be removed. It is what I cleaned up with my last PR https://github.com/Azure/azure-signalr/pull/672